File: backend/internal/models/Review.go
package models

import "time"

type Review struct {
	Id                int               `json:"id"`
	Rating            int               `json:"rating"`
	Client            Client            `json:"client"`
	SpecialistService SpecialistService `json:"specialist_service"`
	Description       string            `json:"description"`
	CreatedAt         time.Time         `json:"created_at"`
}

type ReviewRequest struct {
	Rating              int    `json:"rating"`
	Description         string `json:"description"`
	ClientId            int    `json:"client_id"`
	SpecialistId        int    `json:"specialist_id"`
	SpecialistServiceId int    `json:"specialist_service_id"`
	VisitId             int    `json:"visit_id"`
}

File: frontend/src/models/Review.ts
import {Client} from "./Client";
import {SpecialistService} from "./SpecialistService";

export interface Review {
    id: number
    rating: number
    client: Client
    specialist_service: SpecialistService
    description: string
    created_at: string
}

export interface ReviewRequest {
    rating: number
    description: string
    client_id: number
    specialist_id: number
    specialist_service_id: number
    visit_id: number
}